src/test/java/duke/core/ParserTest.java

```diff
@@ -0,0 +1,64 @@
+package duke.core;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import org.junit.jupiter.api.Test;
+
+import duke.commands.AddCommand;
+import duke.commands.InvalidCommandException;
+import duke.tasks.TaskList;
+
+
+public class ParserTest {
```

Comment:
Very detailed test cases provided! Do consider adding javadocs to your code to improve readability 😄